src/webview-js/toc-editor/toc-editor.jsx

```diff
@@ -0,0 +1,155 @@
+import { h, Fragment, render } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+import 'react-sortable-tree/style.css';
+import SortableTree from 'react-sortable-tree';
+
+const vscode = acquireVsCodeApi();
+const nodeType = 'toc-element'
+
+const removeExpanded = (obj) => {
+  if (typeof obj !== 'object') {
+    return obj
+  }
+  if (obj instanceof Array) {
+    const copy = []
+    for (const subobj of obj) {
+      copy.push(removeExpanded(subobj))
+    }
+    return copy
+  }
+  const { expanded, ...copy } = obj
+  copy.children = removeExpanded(copy.children)
+  return copy
+}
+
+const ContentTree = (props) => {
+  const modifiesStateName = props.modifiesStateName
+  const [data, setData] = useState(props.data);
+
+  const handleChange = (newChildren) => {
+    const { treesData, selectionIndices } = vscode.getState()
+
+    const newData = { ...data, ...{children: newChildren} }
+    const oldStructure = JSON.stringify(removeExpanded(data.children))
+    const newStructure = JSON.stringify(removeExpanded(newChildren))
+
+    if (data.children.length - newChildren.length > 3) {
+      // There's a bug that deletes the whole tree except for one element.
+      // Prevent this by not allowing high magnitude deletions
```

Comment:
:scream: what is this bug?

Reply:
Bug in the dnd tree library. Occurs when you add a duplicate module as the first element in the tree. No idea why it occurs though.